scripts/framework_dev/c_parsing_helper.py

```diff
@@ -0,0 +1,131 @@
+"""Helper functions to parse C code in heavily constrained scenarios.
```

Comment:
The code in `c_parsing_helper.py`, `c_wrapper_generator.py` and `generate_psa_wrappers.py` has not been reviewed. (We rushed these files in via the restricted repository as part of a large security fix.) We need to review them, and most of the code needs to be [reviewed in the next few weeks for the client-server testing project](https://github.com/Mbed-TLS/mbedtls/issues/8961).

Moving the code to a new repository would be a good opportunity to review that code since it'll appear as new in the GitHub interface. For the review and rework to be manageable, these files should be in their own pull request.

Cc @minosgalanakis